### ISSUE TITLE
complete the migration of sgdata pages (upload, etc.)

### ISSUE THREAD

**Issue by denislepage** [MotusDev/Motus-TO-DO#314]
I would like to redirect all users to Motus instead of the remaining sgdata pages, and only keep the SG data API:

https://sgdata.motus.org/download/

The only (?) functionality remaining to transfer are the plots, I believe.

1. The status page I think can now entirely be accessed on our site, correct?

2. Can we confirm that the upload tool is now working consistently on our web site. If so, this can be turned off: https://sgdata.motus.org/upload/

3. The plots are all available under sgm/products. I created a new servlet that checks for permissions at the project level, and then shows users the list of receivers contained in a project directory or files for a specific receiver.

For instance: https://sandbox.motus.org/data/project/plots?projectID=1

There’s a link on the project landing page called Receiver plots. It’ll need a little bit of TLC, but it’s mostly functional. 

- [x]  I would like to add the breadcrumbs to return to “Plots” when I am looking at one specific receiver, like this:
	Manage data › Project "AlaskaBats" (#75) › Plots › SG-3214BBBK8447

- [x]	IOUtils.copy was previoulsy complaining that “java.lang.IllegalStateException: getOutputStream() has already been called for this response”, but the error was not consistent. I think I have now fixed the problem, but if you see the error, let me know. 

Once we confirm that our replacements are all working, and beta7 is launched, we can likely place a redirection note on the sgdata pages.

**Comment by denislepage** [comment]
For some reason, this Lotek receivers (Lotek-D000438) doesn't show any files, even though there are some in the sgm/products folder:
https://sandbox.motus.org/data/project/plots?projectID=1&recvID=2281

All others I have tried so far are working, including other Lotek ones. Puzzled. I thought it could be a case issue, but it doesn't seem that way. It is the folder exists, but it finds no files.

**Comment by zcrysler** [comment]
The email confirmation also includes links to the status page so that should be updated to redirect back to the upload page and the specific log of that job.

The Motus upload tool has been working well for tags and receiver data, but I would make a few text changes:

- https://motus.org/data: it should say "submit tag registrations and detection data"
- https://motus.org/data/project/sgJobs?projectID=116 should have another upload option for tags "a .zip, .7z, or .rar archive containing all the .wav tag recordings and tagreg.txt file for one registration package".

I agree that the list of receivers should have deployment names, most users tell me about issues based off of names, not serial numbers.

**Comment by denislepage** [comment]
Adding names should be easy, though only deployments have name. I guess we show the most recent?

I would expect we should be able to link to this page from other places too.

**Comment by zcrysler** [comment]
I think showing only the most recent will add confusion, users might not know how to find old deployments. I'd say we have to show all deployments or none

**Comment by denislepage** [comment]
Within a receiver folder, file are currently named in a way that don’t always align with our own metadata, because they are entered independently. I therefore can’t easily link a specific deployment to a set of files.

The short project name may come out of database (?), but I suspect the site name comes out of the manifest file on SG, and may not be in totally sync in the metadata. Even so, the names may change, so they are not reliable as way to filter.

Two options:


1.       We display all files for receiver projects associated with a user, like is done on sgdata now, unless a receiver number is specified (e.g. linked directly from elsewhere). They would likely be sorted by receiver number and file names within.



2.       We could also keep the existing structure, but list the deployments (or site names, if those are always used?) of that project besides. You couldn’t click on a deployment or site to only see those associated files, but at least they would be listed. Unique site names would likely be a better approach than deployments, and receivers could even be repeated multiple times if they were deployed at more than 1 site, allowing to show the list in site name order. The only minor downside is that when you clicked on a site name, you would also see other files associated with that receiver from elsewhere.

I think I like option 2 better.

**Comment by zcrysler** [comment]
I think it would be fine if clicking on a site name would show files associated with the receiver from elsewhere, the main thing is that users will want to search for a plot based on name rather than serial number. Option 2 works for me

**Comment by cjardine-bsc** [comment]
FYI that yes, the short project name does come from the database, it should coincide with project_code from the projects table.

**Comment by denislepage** [comment]
Yes, except when it no longer does (that might be rare). Can people change this themselves? In any case, the file names are meaningful to people, which is what matters.

**Comment by denislepage** [comment]
The new pages now seem to work. I have added 2 new links on the project landing page

- Tag sqlite databases  List of tag sqlite databases available for this project
- Receiver plots  List of receiver plots available for this project

Linking to:
/data/project/tagdb?projectID=1
/data/project/plots?projectID=1

We probably should add those links to other places, and also change the notifications on the SGdata side once launched in full. I considered making those links available under downloads, but they are primarily (all?) csv file downloads at the moment, and I wanted to avoid having too many layers deep. We could think about organizing the landing page links a bit more.

Notes: I ran into file permission issues on the NAS drive that made a few folders unavailable initially.  I think that the permissions were initially set on individual receiver folders in sgm/products, but then new folders were not granted sufficient permissions for Windows users. I have now changed the inheritance rules, and also used this to ensure permission inheritance of individual folders:

ICACLS <foldername> /inheritance:e

Hopefully, the problem will be resolved for new folders, but if you see missing files that you know are otherwise available on sgdata, we'll need to look further into it.
